cunumeric/config.py

```diff
@@ -242,3 +245,137 @@ class CuNumericTunable(IntEnum):
     NUM_PROCS = _cunumeric.CUNUMERIC_TUNABLE_NUM_PROCS
     MAX_EAGER_VOLUME = _cunumeric.CUNUMERIC_TUNABLE_MAX_EAGER_VOLUME
     HAS_NUMAMEM = _cunumeric.CUNUMERIC_TUNABLE_HAS_NUMAMEM
+
+
+# Match these to fftType in fft_util.h
+@unique
+class FFTCode(IntEnum):
```

Comment:
This entire class looks like a C code. You could make this object oriented so that the if conditions you're checking in the methods and properties are reflected in the class hierarchy. For example, you could make a class for each FFT type (say R2C, C2R, C2C, D2Z, Z2D, and Z2Z) and have them implement the same set of methods and properties. Then each will need to return the right value without checking anything. For instance, `R2C.input_dtype` would just return `np.float32` unconditionally. Of course, you will still need constructor methods like `real_to_complex_code` or `complex_to_real_code`, but you can eliminate the need for checking the FFT codes once you created them.